modules/services/espanso.nix

```diff
@@ -125,9 +163,16 @@ in
           The services.espanso module only supports Espanso version 2 or later.
         '';
       }
+      {
+        assertion = isLinux -> (cfg.x11Support || cfg.waylandSupport);
+        message = ''
+          In services.espanso at least one of x11 or wayland support must be enabled on linux.
+        '';
+      }
```

Comment:
This isn't true, is it? You have a if else condition that falls back to the original behavior of using the normal package.

Reply:
If the user (accidentally?) sets both to be false then it is better to advice the user, it makes no sense to disable both x11 and wayland.

If we wish to remove this warning. The overall expression does default to the wayland version if both are set to false. (`cfg.package` resolves to `pkgs.espanso-wayland` and `package-bin` resolves to `cfg.package`)

 but I think it is best to leave it in because its an invalid state.

Reply:
We shouldn't be changing the default package to wayland version if it doesn't support X11, that'd be an unexpected breaking change. You added a wayland package specifically for that reason.

Reply:
This isn't a warning, it's an assertion. Which means the eval cannot continue, but there's no reason for it not to since we handle the configuration fine. If both new options are set to false, it should default to the legacy behavior and issue a warning, imo.

Reply:
yeah meant to say assertion. and since it is an invalid state to be in disabling both x11 and wayland support IF we are on linux then better to break eval